src/Features/CSharp/Portable/UsePatternMatching/CSharpAsAndNullCheckDiagnosticAnalyzer.cs

```diff
@@ -39,7 +39,10 @@ public CSharpAsAndNullCheckDiagnosticAnalyzer()
 
         protected override void InitializeWorker(AnalysisContext context)
             => context.RegisterSyntaxNodeAction(SyntaxNodeAction,
-                SyntaxKind.IfStatement, SyntaxKind.WhileStatement);
+                SyntaxKind.IfStatement,
+                SyntaxKind.WhileStatement,
+                SyntaxKind.ReturnStatement,
+                SyntaxKind.LocalDeclarationStatement);
```

Comment:
💭 Seems like this is still missing some (other places where expressions could appear). Not a problem IMO, but I'm thinking at the end of this sequence of PRs we should create an issue to describe the remaining work that would expand this to support "expressions" more generally.

Reply:
Filed https://github.com/dotnet/roslyn/issues/23375